clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml

```diff
@@ -27,13 +27,116 @@
       MatchMode: Full
       RuleId: "remove_cuda"
 
-- Rule: rule_find_package
+- Rule: rule_cuda_no_args_find_package
   Kind: CMakeRule
   Priority: Fallback
-  CmakeSyntax: find_package
-  In: find_package(CUDA${opts})
+  CmakeSyntax: cuda_no_args_find_package
+  MatchMode: Partial
+  In: find_package(CUDA)
+  Out: find_package(IntelSYCL)
+
+- Rule: rule_cuda_find_package
+  Kind: CMakeRule
+  Priority: Fallback
+  CmakeSyntax: cuda_find_package
+  MatchMode: Partial
+  In: find_package(CUDA ${args})
   Out: find_package(IntelSYCL REQUIRED)
 
+- Rule: rule_cub_no_args_find_package
+  Kind: CMakeRule
+  Priority: Fallback
+  CmakeSyntax: cub_no_args_find_package
+  MatchMode: Partial
+  In: find_package(CUB)
+  Out: find_package(oneDPL)
+
+- Rule: rule_cub_find_package
+  Kind: CMakeRule
+  Priority: Fallback
+  CmakeSyntax: cub_find_package
```

Comment:
I'd prefer find_package as the base name for the rule.  eg.  find_package-cub, finally this name will be used for end user if some rule is conflicted against each other.